packages/cli/src/error_handler.ts

```diff
@@ -0,0 +1,81 @@
+import { COLOR, Printer } from '@aws-amplify/cli-core';
+import { InvalidCredentialError } from './error/credential_error.js';
+import { EOL } from 'os';
+import { Argv } from 'yargs';
+
+/**
+ * Attaches process listeners to handle unhandled exceptions and rejections
+ */
+export const attachUnhandledExceptionListeners = (): void => {
+  process.on('unhandledRejection', (reason) => {
```

Comment:
is this intended to be called once per process life? If yes should this aspect be safe guarded somehow?

Reply:
Yeah, the two ways I can think of to enforce this are either use `process.once` or call `process.exit()` in the callback. I think the latter is better because if there are actually multiple unhandled rejections, the former would let all but the first one through whereas the latter would shut down everything after the first one. This is more in line with fail fast and we can probably adjust as needed if there's some other case we need to cover later.

Reply:
That's a good angle to look at too - imo we should handle and log all unexpected rejections.
But I meant this: is it legal to call `attachUnhandledExceptionListeners()` multiple times ?

Reply:
Oh I see. Yeah I can make calling this function idempotent.

Reply:
> imo we should handle and log all unexpected rejections.

That would mean that we leave this as-is. ie the rejection handler would be called for as many unhandled rejections as the process throws